hibernate-envers/src/main/java/org/hibernate/envers/internal/revisioninfo/DefaultRevisionInfoGenerator.java

```diff
@@ -54,9 +57,19 @@ public DefaultRevisionInfoGenerator(
 		this.sessionCacheCleaner = new SessionCacheCleaner();
 	}
 
+	@Override
+	public void setRevisionInfoNumberReader(RevisionInfoNumberReader revisionInfoNumberReader) {
+		this.revisionInfoNumberReader = revisionInfoNumberReader;
+	}
+
 	@Override
 	public void saveRevisionData(Session session, Object revisionData) {
 		session.save( revisionInfoEntityName, revisionData );
+		if ( revisionInfoNumberReader != null ) {
+			if ( revisionInfoNumberReader.getRevisionNumber( revisionData ).longValue() < 0 ) {
```

Comment:
it seems using `&&` is more in line with existing coding pattern, though minor.

Reply:
No worries, I'll make the minor adjustment.  Ideally I would prefer to get rid of the not-null check anyway, but that would require we pass the argument into the constructor, but I wanted to avoid superflous changes.